### PR TITLE
Add Couchbase Integration Back

### PR DESCRIPTION
**Pull request by schmikei**
These values were switched at some point, this should resolve them.

IMPORTANT EDIT: This also adds back couchbase and fixes the integration tests.

Things to validate before this is merged:

- [x] For the memory metric state label, the underlying prometheus metric no longer is present in future versions of couchbase. See https://docs.couchbase.com/server/current/metrics-reference/data-service-metrics.html
- [x] `workload.googleapis.com/couchbase.bucket.error.oom.count` had a missing label in the metadata.yaml
- [x] Certain Metrics value_types needed to be changed from `DOUBLE` => `INT64`
   - `couchbase.bucket.error.oom.count`
   - `couchbase.bucket.operation.count`
- [x] GAUGE => CUMULATIVE  was missed in metadata.yaml, fixed in upcoming PR 
- [x] Change `bucket` metric label to `bucket_name`

### CODE REVIEW

**Comment by schmikei** [comment]
@qingling128 I think this brings things in line for couchbase, we can look over this next week, but kokoro appears to have out of sync metric descriptors. I believe what the `metadata.yaml` that this PR has is what was expected but would you mind double checking just to be sure? Thanks again, I really appreciate it!

File: apps/couchbase.go [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+						otel.RenameLabel("bucket", "bucket_name"),
+					),
+					otel.RenameMetric(
+						"kv_vb_curr_items",
+						"couchbase.bucket.item.count",
+						otel.RenameLabel("bucket", "bucket_name"),
+					),
+					otel.RenameMetric(
+						"kv_num_vbuckets",
+						"coucbhase.bucket.vbucket.count",
```
Comment (ridwanmsharif) [inline apps/couchbase.go:92]: Couchbase is misspelled here

File: integration_test/third_party_apps_data/applications/couchbase/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+expected_metrics:
+  - type: workload.googleapis.com/couchbase.bucket.operation.count
+    value_type: DOUBLE
+    kind: CUMULATIVE
+    monitored_resource: gce_instance
+    labels:
+      bucket_name: .*
+  - type: workload.googleapis.com/couchbase.bucket.item.count
+    value_type: DOUBLE
+    kind: CUMULATIVE
```
Comment (ridwanmsharif) [inline integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:53]: The test says this is implemented as a GAUGE. The scope doc says that too actually, does this need to be updated?

https://docs.google.com/document/d/1_1Iyk6z-1D5EiYuvBdg4_PTWzDa7ovmdnrZMlBS1oPY/edit?usp=sharing

File: integration_test/third_party_apps_data/applications/couchbase/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    labels:
+      bucket_name: .*
+  - type: workload.googleapis.com/couchbase.bucket.memory.usage
+    value_type: DOUBLE
+    kind: GAUGE
+    labels:
+      bucket_name: .*
+    monitored_resource: gce_instance
+    representative: true
+  - type: workload.googleapis.com/couchbase.bucket.item.ejection.count
```
Comment (ridwanmsharif) [inline integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:70]: The test says this is implemented as a GAUGE. The scope doc says that too actually, does this need to be updated?

https://docs.google.com/document/d/1_1Iyk6z-1D5EiYuvBdg4_PTWzDa7ovmdnrZMlBS1oPY/edit?usp=sharing

File: integration_test/third_party_apps_data/applications/couchbase/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    kind: CUMULATIVE
+    monitored_resource: gce_instance
+    labels:
+      bucket_name: .*
+  - type: workload.googleapis.com/couchbase.bucket.vbucket.count
+    value_type: DOUBLE
+    kind: CUMULATIVE
+    monitored_resource: gce_instance
+    labels:
+      bucket_name: .*
```
Comment (ridwanmsharif) [inline integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:62]: Shouldn't there also be a `state` label here?

File: integration_test/third_party_apps_data/applications/couchbase/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  - type: workload.googleapis.com/couchbase.bucket.item.count
+    value_type: DOUBLE
+    kind: CUMULATIVE
+    monitored_resource: gce_instance
+    labels:
+      bucket_name: .*
```
Comment (ridwanmsharif) [inline integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:56]: Shouldn't there also be a `state` label here?

File: integration_test/third_party_apps_data/applications/couchbase/metadata.yaml [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    monitored_resource: gce_instance
+    representative: true
+  - type: workload.googleapis.com/couchbase.bucket.item.ejection.count
+    value_type: DOUBLE
+    kind: GAUGE
+    monitored_resource: gce_instance
+    labels:
+      bucket_name: .*
+  - type: workload.googleapis.com/couchbase.bucket.error.oom.count
```
Comment (ridwanmsharif) [inline integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:76]: For some reason, the test failed and this metric was not reported in the test project - do we know why?

Same thing for:

`workload.googleapis.com/couchbase.bucket.vbucket.count`
`workload.googleapis.com/couchbase.bucket.operation.count`
`workload.googleapis.com/couchbase.bucket.error.oom.count`

Comment (schmikei) [reply integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:78]: Actively looking into these: working in my test environment 
image

Re-running kokoro to see if I can find anything more

**Comment by schmikei** [comment]
@ridwanmsharif are you sure about the resetting of couchbase descriptors for the kokoro project?

Currently still seeing this error in the otel logs 

```
Jul  7 18:18:03 github-test-20220707-5c680-14c11534-3f76-4fd6-8b8a-958a4874ce48 otelopscol[3989]: 2022-07-07T18:18:03.005Z#011error#011exporterhelper/queued_retry.go:149#011Exporting failed. Try enabling retry_on_failure config option to retry on retryable errors#011{"error": "failed to export time series to GCM: rpc error: code = InvalidArgument desc = One or more TimeSeries could not be written: Value type for metric workload.googleapis.com/couchbase.bucket.oom.count conflicts with the existing value type (DOUBLE).\nerror details: name = Unknown  desc = total_point_count:26  success_point_count:25  errors:{status:{code:3}  point_count:1}"}

```

**Comment by ridwanmsharif** [comment]
That is very weird. Taking a look. When attempting to delete it again I get:
```
{
  "error": {
    "code": 404,
    "message": "The metric 'workload.googleapis.com/couchbase.bucket.oom.count' does not exist.",
    "status": "NOT_FOUND",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.DebugInfo",
        "detail": "The metric 'workload.googleapis.com/couchbase.bucket.oom.count' does not exist."
      }
    ]
  }
}
```

So it does look like it was deleted the first time around.

**Comment by schmikei** [comment]
Hmm interesting, I am not a complete expert on how kokoro picks the project, but is there any chance it was a different project do you think? I'll re-run with the op label to see if we get the same error

**Review by sophieyfang — APPROVED** [review]

**Review by ridwanmsharif — APPROVED** [review]

File: integration_test/third_party_apps_data/applications/couchbase/metadata.yaml [hunk 7]
```diff
@@ -0,0 +1,114 @@
+app_url: "https://www.couchbase.com/"
+short_name: couchbase
+long_name: Couchbase
+description: |-
+  The Couchbase integration collects bucket metrics such as operations, memory usage, and ejections. The integration collects metrics from the prometheus server exposed on a node.
```
Comment (mtabasko) [inline integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:5]: Prometheus

File: integration_test/third_party_apps_data/applications/couchbase/metadata.yaml [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+    fields:
+      - name: jsonPayload.message
+        type: string
+        description: "Log message"
+      - name: jsonPayload.client_ip
+        type: string
+        description: The ip address of the client invoking the HTTP request
+      - name: jsonPayload.user
+        type: string
+        description: The name of the user making the http request if basic auth is used.
```
Comment (mtabasko) [inline integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:29]: HTTP

File: integration_test/third_party_apps_data/applications/couchbase/metadata.yaml [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+        type: integer
+        description: The status code of the response of the HTTP request
+  - log_name: couchbase_goxdcr
+    fields:
+      - name: jsonPayload.message
+        type: string
+        description: "Log message"
+      - name: log_type
+        type: string
+        description: The name of the component that is issuing the cross datacenter log
```
Comment (mtabasko) [inline integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:43]: cross-datacenter

File: integration_test/third_party_apps_data/applications/couchbase/metadata.yaml [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+    labels:
+      bucket_name: .*
+    monitored_resource: gce_instance
+configuration_options:
+  metrics:
+    - type: couchbase
+      fields:
+        - name: type
+          default: null
+          description: This value must be couchbase.
```
Comment (mtabasko) [inline integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:104]: This value must be `couchbase`.

File: integration_test/third_party_apps_data/applications/couchbase/metadata.yaml [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+        - name: endpoint
+          default: http://localhost:8091
+          description: The URL of the node to monitor.
+        - name: collection_interval
+          default: 60s
+          description: A time.Duration value, such as 30s or 5m.
```
Comment (mtabasko) [inline integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:110]: A [`time.Duration`](https://pkg.go.dev/time#ParseDuration) value, such as `30s` or `5m`.

File: integration_test/third_party_apps_data/applications/couchbase/metadata.yaml [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+        type: string
+        description: The name of the module issuing the entry
+  - log_name: couchbase_http_access
+    fields:
+      - name: jsonPayload.message
+        type: string
+        description: "Log message"
+      - name: jsonPayload.client_ip
+        type: string
+        description: The ip address of the client invoking the HTTP request
```
Comment (mtabasko) [inline integration_test/third_party_apps_data/applications/couchbase/metadata.yaml:26]: IP

**Review by mtabasko — CHANGES_REQUESTED** [review]
A few minor editorial fixes; these are all things I will have to fix in the generated documentation otherwise.

**Comment by ridwanmsharif** [comment]
Will merge if/when @mtabasko green lights the docs changes.

**Review by mtabasko — APPROVED** [review]
Looks OK from my end. Thanks.

**Pull request merged**